id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java

```diff
@@ -156,6 +155,62 @@ public ResponseWrapper<CredentialIssueResponse> createCredentialIssuance(Credent
 		return credentialIssueResponseWrapper;
 	}
 
+	@Override
+	public ResponseWrapper<CredentialIssueResponse> createCredentialIssuanceByRid(CredentialIssueRequestDto credentialIssueRequestDto, String rid) {
+		LOGGER.debug(IdRepoSecurityManager.getUser(), CREDENTIAL_SERVICE, CREATE_CREDENTIAL,
+				"started creating credential");
+		List<ServiceError> errorList = new ArrayList<>();
+		ResponseWrapper<CredentialIssueResponse> credentialIssueResponseWrapper = new ResponseWrapper<CredentialIssueResponse>();
+
+		CredentialIssueResponse credentialIssueResponse = null;
+		try{
+			CredentialEntity credential=new CredentialEntity();
+			credential.setRequestId(rid);
+			credential.setRequest(mapper.writeValueAsString(credentialIssueRequestDto));
+			credential.setStatusCode(CredentialStatusCode.NEW.name());
+			credential.setCreateDateTime(LocalDateTime.now(ZoneId.of("UTC")));

```

Comment:
You can use `io.mosip.kernel.core.util.DateUtils.getUTCCurrentDateTime()`.

Reply:
taken care